### PR TITLE
Allow the definition of the `_midlSAFEARRAY(HRESULT)` type.

### PR DESCRIPTION
**Pull request by davidschranz**
Pull request for changes described in https://github.com/enthought/comtypes/issues/668

### CODE REVIEW

File: comtypes/safearray.py [hunk 1]
```diff
@@ -123,6 +124,15 @@ def create(cls, value, extra=extra):
...
             numpy arrays must be passed.
             """
+            if cls._vartype_ == VT_HRESULT:
+                raise TypeError(
+                    # There are COM type libraries that define the
+                    # `_midlSAFEARRAY(HRESULT)` type; however, creating `HRESULT`
+                    # safearray pointer instance does not work.
+                    # See also: https://github.com/enthought/comtypes/issues/668
+                    "Cannot create SAFEARRAY type VT_HRESULT."
+                    )
```
Comment (junkmd) [inline comtypes/safearray.py:134]: Formatting issue.

```diff
-                    )
+                )
```

**Review by junkmd — CHANGES_REQUESTED** [review]
Thank you for contributing.

We need to write test code as well.

The test will include assertions ensuring that the definition of `_midlSAFEARRAY(HRESULT)` does not cause an error, while calling `_midlSAFEARRAY(HRESULT).create` raises a `TypeError`.

There are multiple modules for testing safearray, so I am currently looking for the most appropriate place to add the test.

**Review by junkmd — CHANGES_REQUESTED** [review]
I believe that `test/test_midl_safearray_create.py` is the most suitable location to test this behavior.

You could add a test like the following: 

https://github.com/enthought/comtypes/blob/fb15f91d12aedfff1782150ff8e6bc02da4c8257/comtypes/test/test_midl_safearray_create.py#L1-L4

```diff
 # coding: utf-8

-from ctypes import c_int, pointer, POINTER
+from ctypes import c_int, pointer, HRESULT, POINTER
 import unittest
```

https://github.com/enthought/comtypes/blob/fb15f91d12aedfff1782150ff8e6bc02da4c8257/comtypes/test/test_midl_safearray_create.py#L82-L87

```diff
                 self.assertEqual(unpacked.answer, 42)
                 self.assertEqual(unpacked.needs_clarification, True)

+    def test_HRESULT(self):
+        hr = HRESULT(1)
+        sa_type = comtypes.safearray._midlSAFEARRAY(HRESULT)
+        with self.assertRaises(TypeError):
+            sa_type.create([hr], extra=None)
+        with self.assertRaises(TypeError):
+            sa_type.create([hr])
+
     def test_ctype(self):
         extra = None
         cdata = c_int(1)
```

**Review by junkmd — APPROVED** [review]
Thanks.

**Comment by davidschranz** [comment]
Thank you very much for your help!

**Comment by junkmd** [comment]
You're welcome.
This time, the changes to the production code were minimal, and the tests were also kept to a minimum.
Above all, your excellent reproducer and your responsiveness when I asked for verification made this PR much easier to merge.

**Pull request merged**